src/main/java/org/openstreetmap/atlas/geography/boundary/CountryBoundaryMap.java

```diff
@@ -1125,8 +1134,10 @@ public int size()
         List<Polygon> candidates = this.query(target.getEnvelopeInternal());
 
         // Performance improvement, if only one polygon returned no need to do any further
-        // evaluation.
-        if (isSameCountry(candidates))
+        // evaluation (except when geometry has to be sliced at all costs)
+        if (isSameCountry(candidates)
+                && (sources == null || sources.length == 0 || this.forceSlicingPredicate == null
+                        || !this.forceSlicingPredicate.test(sources[0])))
```

Comment:
Two things: 
- Only the first element in `sources` is being used. This might be intentional, but it could be source of confusion.
- If predicate is added to stop slicing operation, then I'd suggest reverting it's conditional, applying it once before collecting candidates for slice operation - right where method starts. If it fails to pass, do not slice at all.

Reply:
Thanks for the comments! The multiple sources come from the use of `Taggable...` which allows me to make it totally optional and ensure I do not break the current API... I can rename it to source though.

According to changing the logic of that code, I am not too sure. I'd rather leave the flow as-is for that PR.